aspnetcore/security/anti-request-forgery/samples/6.x/AntiRequestForgerySample/Snippets/Program.cs

```diff
@@ -33,4 +33,19 @@ public static void AddAntiforgeryOptionsJavaScript(WebApplicationBuilder builder
         builder.Services.AddAntiforgery(options => options.HeaderName = "X-XSRF-TOKEN");
         // </snippet_AddAntiforgeryOptionsJavaScript>
     }
+
+    public static void AddAntiforgeryEndpoint(WebApplication app)
+    {
+#region snippet_AddAntiforgeryEndpoint
+        app.UseAuthorization();
+        app.MapGet("antiforgery/token", (IAntiforgery forgeryService, HttpContext context) =>
+        {
+            var tokens = forgeryService.GetAndStoreTokens(context);
```

Comment:
`GetAndStoreTokens` should write a cookie for you (the cookie name is given by https://github.com/dotnet/aspnetcore/blob/3ea008c80d5cc63de7f90ddfd6823b7b006251ff/src/Antiforgery/src/AntiforgeryOptions.cs#L37). You could read that on the client rather than producing another cookie. If you do want to set a different cookie, consider using a cookie builder like so: https://github.com/dotnet/aspnetcore/blob/3ea008c80d5cc63de7f90ddfd6823b7b006251ff/src/Antiforgery/src/AntiforgeryOptions.cs#L37

Reply:
@pranavkm thanks for the review, I always appreciate it 😅😅.

@blowdart we need both tokens in the client. The `GetAndStoreToken` writes one of them to a non-readable JS that the client needs to be configured to flow with the API requests, and it adds another cookie (`RequestToken`) that will flow from the client to APIs as `HTTP Header`.

This endpoint is basically a variant of https://review.docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-7.0&branch=pr-en-us-26386#generate-antiforgery-tokens-with-iantiforgery

Reply:
Ah that's the part I'd forget, one half is already in a cookie you can't access.

Fair enough :)